### PR TITLE
CI(eos_cli_config_gen): Restructure eos_cli_config_gen host vars for `daemon_terminattr`

### PR DESCRIPTION
**Pull request by laxmikantchintakindi**
## Change Summary

Refactor eos_cli_config_gen host vars for `daemon_terminattr`.

## Related Issue(s)

Fixes #https://github.com/aristanetworks/avd-internal/issues/171

## Component(s) name

`arista.avd.eos_cli_config_gen`

## Proposed changes
Refactor eos_cli_config_gen host vars for `daemon_terminattr`.

### User Checklist

- N/A

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [ ] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

File: ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/terminattr-prem-no-ingestkey.yml [hunk 1]
```diff
@@ -1,13 +0,0 @@
-### Daemon TerminAttr
-daemon_terminattr:
-  cvaddrs:
-    - 10.10.10.8:9910
-    - 10.10.10.9:9910
-    - 10.10.10.10:9910
-  cvvrf: mgt
-  cvauth:
-    method: "key"
-    # Testing without key variable even set.
```
Comment (MaheshGSLAB) [inline ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/terminattr-prem-no-ingestkey.yml:10]: i dont see this case when method is set as key but key variable not set

Comment (laxmikantchintakindi) [reply ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/terminattr-prem-no-ingestkey.yml:10]: I think it is not really needed. The difference is just as below:
With key set as `magickey`:
```
 exec /usr/bin/TerminAttr -cvaddr=10.10.10.8:9910,10.10.10.9:9910,10.10.10.10:9910 **-cvauth=key,magickey** -cvvrf=mgt -cvgnmi -disableaaa -cvproxy=http://arista:arista@10.10.10.1:3128 -grpcaddr=mgmt/0.0.0.0:6042 -grpcreadonly -smashexcludes=ale,flexCounter,hardware,kni,pulse,strata -ingestexclude=/Sysdb/cell/1/agent,/Sysdb/cell/2/agent -taillogs=/var/log/messages,/var/log/agents/ -cvconfig -cvsourceintf=Vlan100
```
Without empty key:

```
exec /usr/bin/TerminAttr -cvaddr=10.10.10.8:9910,10.10.10.9:9910,10.10.10.10:9910 -**cvauth=key,** -cvvrf=mgt -cvgnmi -disableaaa -cvproxy=http://arista:arista@10.10.10.1:3128 -grpcaddr=mgmt/0.0.0.0:6042 -grpcreadonly -smashexcludes=ale,flexCounter,hardware,kni,pulse,strata -ingestexclude=/Sysdb/cell/1/agent,/Sysdb/cell/2/agent -taillogs=/var/log/messages,/var/log/agents/ -cvconfig -cvsourceintf=Vlan100
```

We required one more key to test this empty key.

Comment (MaheshGSLAB) [reply ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/terminattr-prem-no-ingestkey.yml:10]: yeah but without knowing the history we should not move this test from molecule. Even this is going to require to improve the coverage

Comment (laxmikantchintakindi) [reply ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/terminattr-prem-no-ingestkey.yml:10]: Added.

**Review by Vibhu-gslab — APPROVED** [review]

**Review by MaheshGSLAB — APPROVED** [review]

File: ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/hosts.yml [hunk 2]
```diff
@@ -10,12 +10,13 @@ test_hosts:
     host1:
     host2:
     host3:
+    host4:
```
Comment (gmuloc) [inline ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/hosts.yml:12]: dont' add host4 + host4_inline_jinja as suggested here by @ClausHolbechArista plesae rename the host4_inline_jinja to host_inline_jinja / host-inline-jinja (be consistent between the two PRs)

https://github.com/aristanetworks/avd/pull/4712#discussion_r1863547321

File: ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/host5/daemon-terminattr.yml [hunk 3]
```diff
@@ -1,3 +1,4 @@
+---
```
Comment (gmuloc) [inline ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/host5/daemon-terminattr.yml:1]: add a comment stating we are testin on prem-token

File: ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/host4/daemon-terminattr.yml [hunk 4]
```diff
@@ -1,4 +1,4 @@
-### Daemon TerminAttr
+---
```
Comment (gmuloc) [inline ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/host4/daemon-terminattr.yml:1]: comment to state we are testing on prem certs

File: ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/host3/daemon-terminattr.yml [hunk 5]
```diff
@@ -1,3 +1,4 @@
+---
 daemon_terminattr:
```
Comment (gmuloc) [inline ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/host3/daemon-terminattr.yml:3]: comment to state we are testing CVaaS auth (cloud)

File: ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/host2/daemon-terminattr.yml [hunk 6]
```diff
@@ -15,10 +20,16 @@ daemon_terminattr:
...
+        method: "key"
+        key: "arista"
+      cvvrf: mgt
+      cvsourceintf: Vlan500
+    - name: DC3
+      cvaddrs:
+        - 10.40.40.1:9910
+      cvauth:
+        method: "token"
+        token_file: "/tmp/tokenDC2"
```
Comment (gmuloc) [inline ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/host2/daemon-terminattr.yml]:
```suggestion
        token_file: "/tmp/tokenDC3"
```

File: ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/host2/daemon-terminattr.yml [hunk 7]
```diff
@@ -1,5 +1,10 @@
-### Daemon TerminAttr
+---
 daemon_terminattr:
+  cvaddrs:
+    - apiserver.arista.io:443
+  cvauth:
+    method: "key"
```
Comment (gmuloc) [inline ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/host2/daemon-terminattr.yml]: you can remove all the quotes around the strings
```suggestion
    method: key
```

(and below)

File: ansible_collections/arista/avd/molecule/eos_cli_config_gen/documentation/devices/host2.md [hunk 8]
```diff
@@ -361,6 +362,26 @@ dhcp relay
...
+### TerminAttr Daemon
+
+#### TerminAttr Daemon Summary
+
+| CV Compression | CloudVision Servers | VRF | Authentication | Smash Excludes | Ingest Exclude | Bypass AAA |
+| -------------- | ------------------- | --- | -------------- | -------------- | -------------- | ---------- |
+| gzip | 10.20.20.1:9910 | mgt | certs,/persist/secure/ssl/terminattr/DC1/certs/client.crt,/persist/secure/ssl/terminattr/DC1/keys/client.key,/persist/secure/ssl/terminattr/DC1/certs/ca.crt | ale,flexCounter,hardware,kni,pulse,strata | /Sysdb/cell/1/agent,/Sysdb/cell/2/agent | False |
+| gzip | 10.30.30.1:9910 | mgt | key,<removed> | ale,flexCounter,hardware,kni,pulse,strata | /Sysdb/cell/1/agent,/Sysdb/cell/2/agent | False |
+| gzip | 10.40.40.1:9910 | mgt | token,/tmp/tokenDC2 | ale,flexCounter,hardware,kni,pulse,strata | /Sysdb/cell/1/agent,/Sysdb/cell/2/agent | False |
+| gzip | apiserver.arista.io:443 | - | key, | ale,flexCounter,hardware,kni,pulse,strata | /Sysdb/cell/1/agent,/Sysdb/cell/2/agent | False |
```
Comment (gmuloc) [inline ansible_collections/arista/avd/molecule/eos_cli_config_gen/documentation/devices/host2.md]: See the field `|key, |` for Auth (because the key is the empty string. It was happening before this PRbecause we are using `hide_password` for the cluster key but not for the `daemon_terminattr.cvauth.key ` in the documentation template. Given that this is a PR only on Terminattr can you please update the doc template in this 
PR (line 39 add hide_password filter?

**Review by gmuloc — CHANGES_REQUESTED** [review]

File: ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/host4/daemon-terminattr.yml (hunk 4 above)
Comment (laxmikantchintakindi) [reply ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/host4/daemon-terminattr.yml:1]: Done

File: ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/host3/daemon-terminattr.yml (hunk 5 above)
Comment (laxmikantchintakindi) [reply ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/host3/daemon-terminattr.yml:3]: Done

File: ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/host2/daemon-terminattr.yml (hunk 6 above)
Comment (laxmikantchintakindi) [reply ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/host2/daemon-terminattr.yml]: Done

File: ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/host2/daemon-terminattr.yml (hunk 7 above)
Comment (laxmikantchintakindi) [reply ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/host2/daemon-terminattr.yml]: Removed for all.

File: ansible_collections/arista/avd/molecule/eos_cli_config_gen/documentation/devices/host2.md (hunk 8 above)
Comment (laxmikantchintakindi) [reply ansible_collections/arista/avd/molecule/eos_cli_config_gen/documentation/devices/host2.md]: Added.

File: ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/host5/daemon-terminattr.yml [hunk 9]
```diff
@@ -0,0 +1,13 @@
...
```
Comment (gmuloc) [inline ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/host5/daemon-terminattr.yml:1]:
```suggestion
---
# Test on-prem tokens
```

File: ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/host5/daemon-terminattr.yml [hunk 10]
```diff
@@ -0,0 +1,14 @@
...
+# Test on-prem tokens
+# Test on-prem token
```
Comment (gmuloc) [inline ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/host5/daemon-terminattr.yml]: Hmm Github display issue made me commit this..

```suggestion
# Test on-prem tokens
```

File: ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/host5/daemon-terminattr.yml [hunk 11]
```diff
@@ -0,0 +1,13 @@
+---
+# Test on-prem tokens
```
Comment (gmuloc) [inline ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/host5/daemon-terminattr.yml]:
```suggestion
# Test on-prem token
```

**Review by gmuloc — APPROVED** [review]
